skbio/io/format/embed.py

```diff
@@ -226,17 +226,20 @@ def _objects_to_embed(objs, fh, include_embedding_pointer=True):
                 h5grp.attrs["dim"] = emb.shape[1]
 
             # resize if necessary
-            if i > 0:
+            if i > 0: # we don't need to resize for the first object
                 if include_embedding_pointer:
-                    if (len(arr) + idptr_fh[i - 1]) > max_idsize or (
-                        emb.shape[0] + embptr_fh[i - 1]
-                    ) > max_embsize:
-                        max_idsize = ceil(len(arr) + idptr_fh[i - 1]) * resize_by
-                        max_embsize = ceil(emb.shape[0] + embptr_fh[i - 1]) * resize_by
+                    # if (new ID/embs + old ID/embs) > max_size
```

Comment:
TODO : let's add more documentation to this so that this statement is crystal clear.